### PR TITLE
check resourcenames when creating role

### PR DESCRIPTION
**Pull request by xilabao**
not allow multi resources and multi resource names in policyrule.

### CODE REVIEW

File: pkg/apis/rbac/validation/validation.go [hunk 1]
```diff
@@ -96,6 +96,9 @@ func validatePolicyRule(rule rbac.PolicyRule, isNamespaced bool, fldPath *field.
 	if len(rule.Resources) == 0 {
 		allErrs = append(allErrs, field.Required(fldPath.Child("resources"), "resource rules must supply at least one resource"))
 	}
+	if len(rule.ResourceNames) > 0 && len(rule.Resources) > 1 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("resourceNames"), "resource names cannot apply to multiple resources"))
```
Comment (ncdc) [inline pkg/apis/rbac/validation/validation.go]: I don't think `field.Required` is the appropriate error in this case

**Comment by ncdc** [comment]
cc @liggitt @deads2k

File: pkg/apis/rbac/validation/validation.go (hunk 1 above)
Comment (liggitt) [inline pkg/apis/rbac/validation/validation.go]: this is not correct... you could define a rule that let someone create a service, endpoint, and deployment, all named "my-app"

**Comment by liggitt** [comment]
Specifying a resource name with multiple resources is a valid scenario

**Pull request closed without merging**